### PR TITLE
fix(accordion): disable focus on collapsed accordion item content

### LINKED ISSUE #3429: Accordion content retains tab stops for many items (Toggle is one example)
Accordion content tab stops: closed accordion items may still have tab stops within as content is hidden using visibility rather than display: none. Many HTML elements will honor visibility and be removed from the tab stops, but other will not. Some examples are Carbon Checkbox and Carbon Toggle.

>What did you expect to happen?
It is expected that hidden items are removed from tab order within a page

> What happened instead?
Some items retain their tab index

> What WCAG 2.1 checkpoint does the issue violate?
WCAG 2.4.3

1. Create and accordion with two accordion items
2. Add a Toggle to the first accordion item

At runtime
1. Focus on the first Accordion item title
4. Press Tab
Focus stops on the toggle item

### PR DESCRIPTION
**Pull request by emyarod**
Closes #3429

This PR disallows accordion item content to receive focus when the panel is collapsed. This sets `display` none on accordion content when the panel is collapsed.

Alternatively, we can refactor our `hidden` mixin (or each consumer of this mixin) to account for the situation where a typically visible `input` needs to inherit a hidden visibility rule

#### Changelog

**New**

- animations to preserve transitions when expanding and collapsing accordion items

**Changed**

- replace `height`, `visibility`, and `opacity` changes with `display` rule change to disable focus on collapsed panel content
- target CSS3 pseudo selectors
- storybook wrapper `div` styles to make the injected `div` function like a block level element inside our custom flexbox story wrapper (refs https://github.com/storybookjs/storybook/issues/5707)

#### Testing / Reviewing

Add a Toggle or Checkbox to an accordion item and tab through the accordion. The toggle should not receive focus while the panel is collapsed

### CODE REVIEW

File: packages/react/src/components/Accordion/AccordionItem.js [hunk 1]
```diff
@@ -57,8 +61,15 @@ function AccordionItem({
     }
   }
 
+  const handleAnimationEnd = event => {
```
Comment (joshblack) [inline packages/react/src/components/Accordion/AccordionItem.js]: For consistency could we use function declarations?

**Comment by joshblack** [comment]
@emyarod is the linked issue the right one?

**Comment by joshblack** [comment]
Is this a use-case where the `hidden` attribute could come into play?

**Comment by emyarod** [comment]
sure the approach with `hidden` would be identical, the only difference is instead of setting CSS `display` value we would set the `hidden` attribute. is one more preferable over the other?

File: packages/react/src/components/Accordion/AccordionItem.js [hunk 2]
```diff
@@ -27,9 +27,12 @@ function AccordionItem({
 }) {
   const [isOpen, setIsOpen] = useState(open);
   const [prevIsOpen, setPrevIsOpen] = useState(open);
+  const [animation, setAnimation] = useState('');
   const className = cx({
     [`${prefix}--accordion__item`]: true,
     [`${prefix}--accordion__item--active`]: isOpen,
+    [`${prefix}--accordion__item--collapsing`]: animation === 'collapsing',
```
Comment (asudoh) [inline packages/react/src/components/Accordion/AccordionItem.js]:
```suggestion
    [`${prefix}--accordion__item--${animation}`]: animation,
```

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @emyarod!

**Review by designertyler — APPROVED** [review]

**Comment by asudoh** [comment]
@joshblack Would you want to answer the question from @emyarod? Thanks!

**Comment by joshblack** [comment]
I think you're good as-is! Thanks @emyarod 😄

**Review by dakahn — APPROVED** [review]

**Pull request merged**